src/java/org/apache/cassandra/metrics/DroppedMessageMetrics.java

```diff
@@ -44,6 +45,56 @@ public DroppedMessageMetrics(Verb verb)
         this(new DefaultNameFactory("DroppedMessage", verb.toString()));
     }
 
+    public DroppedMessageMetrics(DefaultNameFactory factory)
```

Comment:
It does not seem to me that there is a need for 3 constructor. We could simply use `DroppedMessageMetrics(Verb)`.
It will avoid the need to retrieve the scope and allow us to use an `EnumMap` that will contains the mapping `Verb` <-> old name. Using a static variable for the map would allow us to avoid recreating the map everytime a `DroppedMessageMetrics` is created.

Reply:
The caveat is the other constructor is used with MetricNameFactory and in resetDroppedMessages which requires the interface. My reasoning was I don't want to change too much things for compatibility things which we can easily remove  later when not needed anymore. WDYT?

Reply:
What I had in mind was that: https://github.com/apache/cassandra/commit/b9fc68f84ed7d95dbb1ffb89410193a6696d2c44